File: typescript/src/service.ts
import { ExchangePayload, getInputValue, getOptionalInputValue } from "openbpt-service-core";
import { manifest } from "./manifest";

// TODO: Edit the run function to match your service's requirements
export default async function run(input: ExchangePayload): Promise<ExchangePayload> {
  const n = getInputValue<number>(input, manifest, "number");
  const on = getOptionalInputValue<number>(input, manifest, "optionalNumber");
  const s = getInputValue<string>(input, manifest, "string");
  const b = getInputValue<boolean>(input, manifest, "boolean");
  const m = getInputValue<string>(input, manifest, "model");

  return {
    addToNumber: on ? n + on : n + 1,
    addExclamationMarkToString: `${s}!`,
    getOppositeBoolean: !b,
    newModel: m,
  };
}

File: python/src/service.py
def run(input_data):
    output_data = {}

    n = input_data.get("number")
    on = input_data.get("optionalNumber")
    s = input_data.get("string")
    b = input_data.get("boolean")
    m = input_data.get("model")

    output_data["addToNumber"] = n + 1 if on is None else n + on
    output_data["addExclamationMarkToString"] = f"{s}!"
    output_data["getOppositeBoolean"] = not b
    output_data["newModel"] = m

    return output_data
